src/sqlfluff/cli/commands.py

```diff
@@ -937,6 +861,223 @@ def fix(
+    help="An optional suffix to add to fixed files.",
+)
+@click.option(
+    "--FIX-EVEN-UNPARSABLE",
+    is_flag=True,
+    default=None,
+    help=(
+        "Enables fixing of files that have templating or parse errors. "
+        "Note that the similar-sounding '--ignore' or 'noqa' features merely "
+        "prevent errors from being *displayed*. For safety reasons, the 'fix'"
+        "command will not make any fixes in files that have templating or parse "
+        "errors unless '--FIX-EVEN-UNPARSABLE' is enabled on the command line"
+        "or in the .sqlfluff config file."
+    ),
+)
+@click.option(
+    "--show-lint-violations",
+    is_flag=True,
+    help="Show lint violations",
+)
+@click.argument("paths", nargs=-1, type=click.Path(allow_dash=True))
+def fix(
+    force: bool,
+    paths: Tuple[str],
+    bench: bool = False,
+    fixed_suffix: str = "",
+    logger: Optional[logging.Logger] = None,
+    processes: Optional[int] = None,
+    disable_progress_bar: Optional[bool] = False,
+    extra_config_path: Optional[str] = None,
+    ignore_local_config: bool = False,
+    show_lint_violations: bool = False,
+    **kwargs,
+) -> None:
+    """Fix SQL files.
+
+    PATH is the path to a sql file or directory to lint. This can be either a
+    file ('path/to/file.sql'), a path ('directory/of/sql/files'), a single ('-')
+    character to indicate reading from *stdin* or a dot/blank ('.'/' ') which will
+    be interpreted like passing the current working directory as a path argument.
+    """
+    # some quick checks
+    fixing_stdin = ("-",) == paths
+
+    config = get_config(
+        extra_config_path, ignore_local_config, require_dialect=False, **kwargs
+    )
+    fix_even_unparsable = config.get("fix_even_unparsable")
+    output_stream = make_output_stream(
+        config, None, os.devnull if fixing_stdin else None
+    )
+    lnt, formatter = get_linter_and_formatter(config, output_stream)
+
+    verbose = config.get("verbose")
+    progress_bar_configuration.disable_progress_bar = disable_progress_bar
+
+    formatter.dispatch_config(lnt)
+
+    # Set up logging.
+    set_logging_level(
+        verbosity=verbose,
+        formatter=formatter,
+        logger=logger,
+        stderr_output=fixing_stdin,
+    )
+
+    # handle stdin case. should output formatted sql to stdout and nothing else.
+    if fixing_stdin:
+        _stdin_fix(lnt, formatter, fix_even_unparsable)
+    else:
+        _paths_fix(
+            lnt,
+            formatter,
+            paths,
+            processes,
+            fix_even_unparsable,
+            force,
+            fixed_suffix,
+            bench,
+            show_lint_violations,
+        )
+
+
+@cli.command(name="format", cls=DeprecatedOptionsCommand)
+@common_options
+@core_options
+@lint_options
+@click.option(
+    "-x",
+    "--fixed-suffix",
+    default=None,
+    help="An optional suffix to add to fixed files.",
+)
+@click.option(
+    "--FIX-EVEN-UNPARSABLE",
```

Comment:
In the spirit of `format` being a simpler, safer command, I wonder if we should consider omitting this option, which is known to be unsafe.